[bindings] Test SimbodyMatterSubsystem::calcResidualForce()

chrisdembia opened the pull request:
This PR just makes sure that the method SimbodyMatterSubsystem::calcResidualForce() and the class OpenSim::InverseDynamicsSolver are accessible from MATLAB/Python.

tkuchida commented:
LGTM :+1:

chrisdembia commented:
Thanks Tom; anyone interested in reviewing?

aymanhab commented:
@chrisdembia This looks great in terms of verifying the right sizes of outputs and that the methods exist, how hard is it to check for actual meaningful values?

tkuchida commented:
> check for actual meaningful return values

Good idea. I don't think a regression test is appropriate here, but it would be nice to ensure all the values aren't `0` or `NaN`, for example.

chrisdembia commented:
I have added more rigorous checks in the tests, and I also added to the CHANGELOG that the SimbodyMatterSubsystem is now available in scripting.

Ready for review again.

humphreysb commented:
@chrisdembia - I will look at it over the weekend.

chrisdembia commented:
Great; thank you.

tkuchida commented on CHANGELOG.md:
```diff
@@ -79,8 +79,14 @@ Removed Classes
+MATLAB and Python interfaces
+----------------------------
+- The SimbodyMatterSubsystem class--which provides operators related to the mass
+matrix, jacobians, inverse dynamics, etc.--is now accessible in MATLAB and
```
jacobians -> Jacobians (named after Carl Gustav Jacob Jacobi).

tkuchida commented:
One minor comment; otherwise, looks good to go :+1:

humphreysb commented on Bindings/Java/Matlab/tests/testSimbody.m:
```diff
@@ -0,0 +1,40 @@
+
+s = model.initSystem();
+
+model.realizeDynamics(s);
```
Looks Good.  One question, aren't you only setting acceleration for one of the coordinates?

aymanhab commented:
After addressing @humphreysb clarification (in code comments) this should be ready to merge, as a significant improvement over what we have. Thanks @chrisdembia

chrisdembia commented:
Thanks @tkuchida, @aymanhab and @humphreysb for comments. Based on @aymanhab's permission, I am merging this.
